sql/core/src/main/scala/org/apache/spark/sql/jdbc/CassandraDialect.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.jdbc
+
+import java.sql.Types
+
+import org.apache.spark.sql.types._
+
+
+private case object CassandraDialect extends JdbcDialect {
+
+  override def canHandle(url: String): Boolean =
+    url.startsWith("jdbc:datadirect:cassandra") ||
+    url.startsWith("jdbc:weblogic:cassandra")
+
+  override def getInsertStatement(table: String, rddSchema: StructType): String = {
+    val sql = new StringBuilder(s"INSERT INTO $table ( ")
+    var fieldsLeft = rddSchema.fields.length
+    var i = 0
+    // Build list of column names
+    while (fieldsLeft > 0) {
+      sql.append(rddSchema.fields(i).name)
+      if (fieldsLeft > 1) sql.append(", ")
```

Comment:
@rxin are you the best person to ask about overloading `DataFrameWriter.jdbc()`?

Interesting question about maintaining the current behavior when no column name mapping is specified. In a way it still seems suboptimal to allow this behavior. What if there are the same number of columns, and all are the same type, but the ordering is different? you'd silently insert the wrong data in the wrong column.

Although specifying the DataFrame-to-table column name mapping can be optional (or, the caller can override only the names they want to) I think the SQL statement should be explicit. It does mean that someone who has a DataFrame with differently-named columns somehow might now encounter an exception, but I wonder if that's actually the right thing to enforce going forward. If it doesn't match up by name, don't proceed.

The API changes will take some care to make sure it's unintrusive and backwards compatible.

I suspect it doesn't do much harm to keep the insert statement logic in `JdbcDialects` though I imagine this behavior, whatever we decide, will be the right thing for all dialects, so it can be a default implementation there.